### PR TITLE
Fixes #22206 - provide after_sync hook

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
After installing foreman_hooks, simple create some bash script in your foreman dir:

config/hooks/katello/repository/after_sync/foo.sh

chmod 755 config/hooks/katello/repository/after_sync/foo.sh

**Comment by jdshewey** [comment]
When I apply this patch, it seems to break /usr/sbin/foreman-rake db:migrate. It seems I can only apply it _after_ install. (You can rake it manually after-the-fact to see the issue though)

File: app/models/katello/repository.rb [hunk 1]
```diff
@@ -681,6 +683,17 @@ def destroyable?
       return true
     end
 
+    def sync_hook
+      run_callbacks :sync do
+        logger.debug "custom hook after_after on #{name} will be executed if defined."
```
Comment (jdshewey) [inline app/models/katello/repository.rb]: Shouldn't this be after_sync?

Comment (jlsherrill) [reply app/models/katello/repository.rb]: good call :)

**Comment by jlsherrill** [comment]
@jdshewey were you ever able to get this to work?  It worked for me in my initial testing.  You will need https://github.com/theforeman/foreman_hooks/commit/ba5ac364b69a29998e4931fa46153794124ef439  for the stdin to work properly. 

My script was in:   config/hooks/katello/repository/after_sync/foo.sh

Which i think on a production install would be /etc/foreman/katello/repository/after_sync/foo.sh

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
[test katello]

**Comment by thomasmckay** [comment]
rake db:migrate 
```
NameError: uninitialized constant Actions::Middleware::KeepCurrentTaxonomies
/home/vagrant/code/foreman-hooks/lib/foreman_hooks/as_dependencies_hook.rb:4:in `load_missing_constant'
/home/vagrant/code/katello/app/lib/actions/katello/upstream_subscriptions/bind_entitlement.rb:5:in `<class:BindEntitlement>'
/home/vagrant/code/katello/app/lib/actions/katello/upstream_subscriptions/bind_entitlement.rb:4:in `<module:UpstreamSubscriptions>'
/home/vagrant/code/katello/app/lib/actions/katello/upstream_subscriptions/bind_entitlement.rb:3:in `<module:Katello>'
/home/vagrant/code/katello/app/lib/actions/katello/upstream_subscriptions/bind_entitlement.rb:2:in `<module:Actions>'
/home/vagrant/code/katello/app/lib/actions/katello/upstream_subscriptions/bind_entitlement.rb:1:in `<top (required)>'
```

**Comment by thomasmckay** [comment]
ACK works for me on fresh install

**Pull request merged**